loopchain/baseservice/tx_message.py

```diff
@@ -0,0 +1,106 @@
+"""helper class for TxMessages"""
+
+import json
+import queue
+import sys
+from typing import List, Optional
+
+from loopchain import configure
+from loopchain.blockchain.transactions import Transaction, TransactionVersioner, TransactionSerializer
+from loopchain.protos import loopchain_pb2
+
+
+class TxItem:
+    tx_serializers = {}
+
+    def __init__(self, tx_json: str, channel: str):
+        self.channel = channel
+        self.__tx_json = tx_json
+        self.__len = sys.getsizeof(tx_json) + sys.getsizeof(channel)
+
+    def __len__(self):
+        return self.__len
+
+    def get_tx_message(self):
+        message = loopchain_pb2.TxSend(
+            tx_json=self.__tx_json,
+            channel=self.channel)
+        return message
+
+    @classmethod
+    def create_tx_item(cls, tx_param: tuple, channel: str):
+        tx, tx_versioner = tx_param
+        tx_serializer = cls.get_serializer(tx, tx_versioner)
+        tx_item = TxItem(
+            json.dumps(tx_serializer.to_raw_data(tx)),
+            channel
+        )
+        return tx_item
+
+    @classmethod
+    def get_serializer(cls, tx: Transaction, tx_versioner: TransactionVersioner):
+        if tx.version not in cls.tx_serializers:
+            cls.tx_serializers[tx.version] = TransactionSerializer.new(tx.version, tx.type(), tx_versioner)
+        return cls.tx_serializers[tx.version]
+
+
+class TxMessages:
+    def __init__(self):
+        self.total_size = 0
+        self._transactions: List[TxItem] = []
+
+    def __len__(self):
+        return len(self._transactions)
+
+    def append(self, tx_item):
+        self._transactions.append(tx_item)
+        self.total_size += len(tx_item)
+
+    def get_messages(self) -> List:
+        tx_messages = [tx.get_tx_message() for tx in self._transactions]
+
+        return tx_messages
+
+    def reset(self):
```

Comment:
How about naming `clear`, to follow `list` protocol in python?

Reply:
removed